src/app/forms/components/dynamic-form-group/dynamic-form-group.component.ts

```diff
@@ -35,7 +35,9 @@ export class DynamicFormGroupComponent implements OnInit {
     // NiallB: This populates data without respecting where in the hierarchy it is (see unit tests)
     Object.keys(data).forEach((dataKey) => {
       if (typeof data[dataKey] === 'object') {
-        this.populateData(data[dataKey], template);
+        if (data[dataKey] !== null) {
```

Comment:
Can we use a truthy evaluation here (can only think of empty string being the issue)?

Also would a similar thing happen in the else block, is this explicitly data[datakey] check or more so on the data key?

Also if the above doesn't apply could this be added on the if statement above as to not reduce our nesting ?

Reply:
This is basically checking that if we had the following:
```
{
  someProperty: 'value'
  somePropertyObject: {}
}
```
...that `somePropertyObject` isn't actually empty because if it is empty we don't want to iterate inside it. So it might be worth adding it to the if statement above like you mentioned. 

The else statement is concerned for when the value of `data[datakey]` isn't an object i.e `someProperty` so I don't believe we would need a null check like we do in the first if statement as we're not looking to iterate over it. If the value of `someProperty` is null then we would only be putting the value of null within `template.value` (which is already `null`/`""`) so it wouldn't really be worth adding a null check here.

Reply:
@tom-bjss & @aaron-downing 
We found we see
```
{
  someProperty: null
}
```

Where `null` passes the `=== object` test but `null` itself doesn't have any keys (so `Object.keys(null).forEach(...)` obviously fails)

But yes, we could do `if (data[dataKey]) { /* do stuff */ }` as a truthy-style guard clause, I just thought the ` !== null` was explicitly clear. Happy to change it  :)